dftb-uv_2d.py

```diff
@@ -200,49 +178,31 @@ def smooth_spectrum(path, min_energy, max_energy, min_wavelength, max_wavelength
         xmin_spectrum = min(0.0, min_energy)
         xmax_spectrum = math.ceil(max_energy) + spectrum_discretization_step
 
-    # global lists
-    statelist = list()  # mode
-    energylist = list()  # energy eV
-    intenslist = list()  # fosc
     gauss_sum = list()  # list for the sum of single gaussian spectra = the convoluted spectrum
 
-    spectrum_file = path + '/' + 'EXC.DAT'
+    spectrum_file = path + '/' + dir + '/' + 'EXC.DAT'
 
     # open a file
     # check existence
     try:
-        with open(spectrum_file, "r") as input_file:
-            count_line = 0
-            for line in input_file:
-                # only recognize lines that start with number
-                # split line into 3 lists mode, energy, intensities
-                # line should start with a number
-                if 5 <= count_line <= 55:
-                    if re.search("\d\s{1,}\d", line):
-                        energylist.append(float(line.strip().split()[0]))
-                        intenslist.append(float(line.strip().split()[1]))
-                else:
-                    pass
-                count_line = count_line + 1
+        energylist, intenslist = read_dftb_output(spectrum_file, read_dftb_outoput_start_line, read_dftb_outoput_end_line)
 
     # file not found -> exit here
     except IOError:
         print(f"'{spectrum_file}'" + " not found", flush=True)
-        sys.exit(1)
+    except Exception as e:
+        print("Rank: ", comm_rank, " encountered Exception: ", e.message, e.args)
 
     try:
-        smile_string_file = path + '/' + 'smiles.pdb'
+        smile_string_file = path + '/' + dir + '/' + 'smiles.pdb'
         mol = MolFromPDBFile(smile_string_file, sanitize=False, proximityBonding=True)
         mol = RemoveAllHs(mol)
         with open(smile_string_file, "r") as input_file:
             smiles_string = MolToSmiles(mol)
     except IOError:
         print(f"'{smile_string_file}'" + " not found", flush=True)
-        sys.exit(1)
     except Exception as e:
-        print("Rank: ", comm_rank, " encountered Exception: ")
-        smiles_string = smile_string_file
-        # comm.Abort(1)
+        print("Rank: ", comm_rank, " encountered Exception: ", e.message, e.args)
```

Comment:
Similarly here and other places - the Exception object does not have a 'message' attribute